build.sbt

```diff
@@ -41,14 +41,28 @@ lazy val allVersionCompilerLintSwitches = Seq(
   "-feature",
   "-unchecked",
   "-Xfatal-warnings",
-  "-Xlint",
   "-Yno-adapted-args",
   "-Ywarn-dead-code"
 )
 
-lazy val newerCompilerLintSwitches = Seq(
+lazy val scala211Flags = Seq(
   "-Ywarn-unused-import", // Not available in 2.10
-  "-Ywarn-numeric-widen" // In 2.10 this produces a some strange spurious error
+  "-Ywarn-numeric-widen" // In 2.10 this produces a strange spurious error
+)
+
+// Scala 2.12.2 has excessive warnings about unused implicits
```

Comment:
That could be it. I didn't search for a bug, I just saw lots of compiler errors when I upgraded to 2.12.2. I saw cats has similar issues; their discussion is here - https://github.com/typelevel/cats/pull/1632/files/92a1989c88fc44d6da145b6e0fee3ec24c099ad5#diff-fdc3abdfd754eeb24090dbd90aeec2ceR574